phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/PhoenixMasterObserver.java

```diff
@@ -0,0 +1,215 @@
+                                               final RegionInfo regionInfoA, final RegionInfo regionInfoB) {
+        Configuration conf = c.getEnvironment().getConfiguration();
+        try {
+            Connection conn  = QueryUtil.getConnectionOnServer(conf);
+            // CDC will be enabled on Phoenix tables only
+            PTable phoenixTable = getPhoenixTable(conn, regionInfoA.getTable());
+            if (phoenixTable == null) {
+                LOGGER.info(regionInfoA.getTable() + " is not a Phoenix Table, " +
+                        "skipping partition metadata update.");
+                return;
+            }
+            // find streamName with ENABLED status
+            String tableName = phoenixTable.getName().getString();
+            PreparedStatement pstmt = conn.prepareStatement(STREAM_STATUS_QUERY);
+            pstmt.setString(1, tableName);
+            ResultSet rs = pstmt.executeQuery();
+            if (rs.next()) {
+                String streamName = rs.getString(1);
+                LOGGER.info("Updating partition metadata for table=" + tableName + ", stream="
+                        + streamName + " for daughters " + regionInfoA.getEncodedName() + " "
+                        + regionInfoB.getEncodedName());
+                String parentPartitionID = getParentPartitionId(conn, tableName, streamName, regionInfoA, regionInfoB);
+                upsertDaughterPartition(conn, tableName, streamName, parentPartitionID, regionInfoA);
+                upsertDaughterPartition(conn, tableName, streamName, parentPartitionID, regionInfoB);
+                updateParentPartitionEndTime(conn, tableName, streamName, parentPartitionID, regionInfoA.getRegionId());
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Unable to update CDC Stream Partition metadata during split with daughter regions: "
+                    + regionInfoA.getEncodedName() + " " + regionInfoB.getEncodedName() );
+            LOGGER.error("Error: " + e);
+        }
+    }
+
+    private PTable getPhoenixTable(Connection conn, TableName tableName) throws SQLException {
+        PTable pTable;
+        try {
+            pTable = PhoenixRuntime.getTable(conn, tableName.toString());
+        } catch (TableNotFoundException e) {
+            return null;
+        }
+        return pTable;
+    }
+
+    /**
+     * Lookup parent's partition id (region's encoded name) in SYSTEM.CDC_STREAM.
+     * We infer start and end key for the parent from the two daughters' start/end keys.
+     */
+    private String getParentPartitionId(Connection conn, String tableName, String streamName,
+                                        RegionInfo regionInfoA, RegionInfo regionInfoB)
+            throws SQLException {
+        byte[] inferredParentStartKey =
+                Bytes.compareTo(regionInfoA.getStartKey(), regionInfoB.getStartKey()) < 0 ?
+                regionInfoA.getStartKey() : regionInfoB.getStartKey();
```

Comment:
I confirmed that regionInfoA.getStartKey() is not expected to be greater than regionInfoB.getStartKey()

Reply:
Probably we can infer parent key directly as regionInfoA startKey and regionInfoB endKey.

Reply:
Javadoc also says that left daughter is A and right daughter is B:

```
  /**
   * Called after the region is split.
   * @param c           the environment to interact with the framework and master
   * @param regionInfoA the left daughter region
   * @param regionInfoB the right daughter region
   */
  default void postCompletedSplitRegionAction(final ObserverContext<MasterCoprocessorEnvironment> c,
    final RegionInfo regionInfoA, final RegionInfo regionInfoB) throws IOException {
  }
```

Reply:
Got it!